sdk/tables/tests.yml

```diff
@@ -6,6 +6,7 @@ jobs:
       BuildTargetingString: azure-data-tables
       ServiceDirectory: tables
       AllocateResourceGroup: 'false'
+      DeployArmTemplate: 'true'
```

Comment:
We should remove the AllocateResourceGroup false here and just by the fact someone set DeployArmTemplate we know not to allocate the resource group. (Similarly in the other tests.yml).

Reply:
For this specific library, both the ResourceGroup and the ArmTemplate are used for testing. I will work to making them all dependent on the ArmTemplate, but there are ~20 test files for Tables that I would need to fix. This will be done in a separate PR unless absolutely adamant about having one or the other.

Reply:
Hmm... so tables is using both the arm template and the allocate resource group?  If that is the case then why is allocate resource group set to false? If there are cases were both are needed then I'm find cleaning them up later, but I think it would be good to get everything using one or the other. In fact the next step could be to completely eliminate the AllocationResoureGroup step and replace it with a common template that only setups up a resource group.

Reply:
Hmm...you're right I was misunderstanding. The Preparers go through and create resource groups/storage accounts.
To your other points, libraries will use one or the other (either the ArmTemplate in combo with the `PowerShellPreparer` or the specialized Preparers for the specific service). I can open tracking issues into what libraries still rely on the AllocateResourceGroup and work on replacing with a common template.